src/RowContainer.js

```diff
@@ -0,0 +1,44 @@
+import React, {PropTypes} from 'react';
+import {ContextMenuLayer} from 'react-contextmenu';
+
+class RowContainer extends React.Component {
+  hasContextMenu() {
+    return this.props.contextMenu && React.isValidElement(this.props.contextMenu);
+  }
+
+  renderRowsWithContextMenu() {
+    let newProps = {rowIdx: this.props.rowIdx, idx: this.props.idx};
+    let contextMenu = React.cloneElement(this.props.contextMenu, newProps);
```

Comment:
This component is a simple wrapper for all rows. So we will only clone once when the canvas is rerendered. Change the name here to `RowsContainer`